test/apps/iodemo/ucx_wrapper.cc

```diff
@@ -378,12 +378,13 @@ ucs_status_t UcxContext::epoll_init()
     return UCS_OK;
 }
 
-void UcxContext::progress_worker_event()
+void UcxContext::progress_worker_event(size_t loop)
 {
     int         ret;
     epoll_event ev;
 
     if (ucp_worker_progress(_worker)) {
+        for (int i = 0; (loop == 0 || i < loop) && ucp_worker_progress(_worker); i++);
```

Comment:
instead of lines 386-389:
```
unsigned i = 0;
do {
  if (++i > count) {
     return; /* exceeded progress count */
  }
} while (ucp_worker_progress(_worker) > 0);
```

Reply:
I just committed a patch addressing all your comments. But for this comment, I have a concern.
The original logic was to go through epoll only if no ucp_worker_progress was made, but with your code,  as long as max count of ucp_worker_progress calls was not reached, epoll is needed which seems not reasonable.
Since the new scheme is to do progress polling in many places instead of only when sending window is closed. no progress being made is OK and should not epoll_wait for next progress which might hinder performance. IMHO, we should remove these epoll codes completely.

Reply:
The latest commit adds
+    if (i > 1)
+        return;
to make sure go through epoll_wait only if no progress is made.